ykcapi/src/lib.rs

```diff
@@ -41,22 +41,22 @@ pub extern "C" fn yk_mt_control_point(_mt: *mut MT, _loc: *mut Location) {
 }
 
 // The "real" control point, that is called once the interpreter has been patched by ykllvm.
+// Returns the address of a reconstructed stack or null if there wasn#t a guard failure.
```

Comment:
wasn#t

Reply:
How does the reconstructed stack get used? It seems that a pointer to it is returned directly to the caller. What does the caller then do with it?

Reply:
This is also explain better in the commit message. The caller checks the value of the return and if it's not null, then it jumps to `__ykrt_reconstruct_frames` which copies over the new stack.